src/components/split-button/split-button-rtl.test.tsx

```diff
@@ -0,0 +1,912 @@
+import React from "react";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent, { UserEvent } from "@testing-library/user-event";
+import StyledButton from "components/button/button.style";
+import SplitButton from "./split-button.component";
+import Button from "../button";
+import { SizeOptions } from "../button/button.component";
+import {
+  testStyledSystemMargin,
+  toCSSCase,
+} from "../../__spec_helper__/test-utils";
+import I18nProvider from "../i18n-provider";
+
+jest.mock("../../__internal__/utils/helpers/guid", () => () => "guid-12345");
+
+const sizes = ["small", "medium", "large"] as const;
+type SizeType = typeof sizes[number];
+
+const singleButton = (size: SizeType = "medium") => (
+  <Button size={size} key="testKey">
+    Single Button
+  </Button>
+);
+
+const multipleButtons = [
+  <Button key="testKey1">Extra Button 1</Button>,
+  <Button key="testKey2">Extra Button 2</Button>,
+  <Button key="testKey3">Extra Button 3</Button>,
+];
+
+type SizeConfig = {
+  fontSize: string;
+  minHeight: string;
+  paddingLeft: string;
+  paddingRight: string;
+};
+const buildSizeConfig = (size: SizeOptions): SizeConfig => {
+  if (size === "small") {
+    return {
+      minHeight: "32px",
+      paddingLeft: "var(--spacing200)",
+      paddingRight: "var(--spacing200)",
+      fontSize: "var(--fontSizes100)",
+    };
+  }
+
+  if (size === "large") {
+    return {
+      minHeight: "48px",
+      paddingLeft: "var(--spacing400)",
+      paddingRight: "var(--spacing400)",
+      fontSize: "var(--fontSizes200)",
+    };
+  }
+
+  return {
+    minHeight: "40px",
+    paddingLeft: "var(--spacing300)",
+    paddingRight: "var(--spacing300)",
+    fontSize: "var(--fontSizes100)",
+  };
+};
```

Comment:
We don't want to use parametrised tests anymore, we need to focus on making our tests readable. I don't mind removing the size config and having them explicitly added in the test cases but it will mean more work if we ever change any values

Reply:
I agree, I see there is potential for this feature to be overused and get out of hand